CriteoPublisherSdk/Sources/Threading/CR_ThreadManager.h

```diff
@@ -39,6 +39,28 @@ NS_ASSUME_NONNULL_BEGIN
 /** Runs with a context interacting with the CR_ThreadManager instance. */
 - (void)runWithCompletionContext:(void (^)(CR_CompletionContext *))block;
 
+typedef void (^dispatchWithTimeoutHandler)(BOOL handled);
+typedef void (^dispatchWithTimeoutOperationHandler)(
+    void (^completionCallback)(dispatchWithTimeoutHandler));
+
+/**
+ * Dispatch asynchronously an operation block and a timeout block concurrently.
+ *
+ * Whatever the outcome of the operation is, both handlers will be called with a flag indicating if
+ * the operation has already been handled.
+ * - When the Operation block completes, it must call the provided block with a callback argument
+ *   that will be called providing the `handled` flag
+ * - When the timeout is reached, the `timeoutHandler` will be called with the `handled` flag.
+ *
+ * @param timeout Timeout value in seconds
+ * @param operationHandler Block to run on Global Queue, with a block callback argument to retrieve
+ * the handled status
+ * @param timeoutHandler Block to be run on timeout, with the handled flag as argument
+ */
+- (void)dispatchAsyncOnGlobalQueueWithTimeout:(NSTimeInterval)timeout
```

Comment:
Thank you for this documentation. It helps to understand what's going on here and what can be expected.
Although, using block of block calling block is quite hard to get. I think it might help if you add a sample usage in the doc.

Reply:
You're welcome, ok will add an example. OTOH, the unit tests are kind of examples, isn't enough?

Reply:
Unit tests are examples but are not always easy to read unfortunately 😞

Reply:
Fixed by b1639d6